flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -1916,6 +1918,50 @@ else if (value instanceof Row) {
 		}
 	}
 
+	public static <X> TypeInformation<X> getForCollection(Iterable<X> value) {
+		return new TypeExtractor().privateGetForIterable(value);
+	}
+
+	public static <X> TypeInformation<X> getForCollection(X[] value) {
+		return new TypeExtractor().privateGetForIterable(Arrays.asList(value));
+	}
+
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	private <X> TypeInformation<X> privateGetForIterable(Iterable<X> value) {
+		checkNotNull(value);
+
+		Iterator<X> it = value.iterator();
+		X v = checkNotNull(it.next());
```

Comment:
Calling `next` without checking can result in `NoSuchElementException`. We should check that first and throw a helpful exception.